### DISCUSSION TITLE
What do Terrains Look Like?

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#349] [Ideas]
There are basically four forms terrain can come in. 

**Single Raster (.tif)**
This means there's a single raster with our terrain elevations. The simplest case

**Multi Raster (.vrt + Xtifs) **
This means there's a single .vrt file that points at a bunch of tifs. 

**RAS Terrain (.hdf +.vrt + Xtifs)**
When RAS stitches together a terrain, it creates an .hdf file. That .hdf contains pointers to all the tiffs that make it up. Always has a .vrt with it too

**RAS Terrain Modified (Master .hdf + supporting .hdf + .vrt + Xtifs**)
With RAS 6.0 we introduced the concept of a terrain modifications. Those modifications are typically saved to what is technically a new terrain, that points at the old one, and saves modification to it in a new one. It doesn't totally replace it though, it still has to know the terrain it's modifying, so there's actually a couple .hdf's in the directory. Thats ok though, The user just has to point at the one they want to use, and I don't think we need to hold their hand in that. Just let them select it.

**Comment by codymccoy** [comment]
the basic rules i have are as follows:
if the selected path is tif or flt then i just copy that individual file into the study. If the selected path is vrt, then i copy the whole directory that the selected file is in. 

Sounds like Brennan is suggesting that there be another option: if selected path is HDF, then copy the whole directory (excluding any other hdf files?) into our study.

That probably isn't too difficult to add. I will wait for Richard to sign off on it.

**Comment by rnugent3** [comment]
Let's plan on talking about this during our biweekly tomorrow.
